### PR TITLE
Lambda Functions - Write the code into the DB

### PR DESCRIPTION
**Pull request by liranmauda**
### Explain the changes
- Writing the code into the DB
- Fix an issue where child process called generate_entropy on a container
- Async await some code

Signed-off-by: liranmauda <liran.mauda@gmail.com>

### Issues:
1. when using postgres db the Function Statistics in the UI is not working (We are missing some MapReduce functionality in postgres, regarding func_stats) 

### Limitation
1. As we are missing func_stats MapReduce for postgres, the func_stats will not be written in the db when using postgres

### Testing Instructions:
1. Test lambda function create/delete/update/trigger
2. Test with mongodb and with postgres

### CODE REVIEW

File: src/server/func_services/func_server.js [hunk 1]
```diff
@@ -162,15 +164,28 @@ async function update_func(req) {
...
 }
 
 async function read_func(req) {
     await _load_func(req);
-    const reply = await _get_func_info(req.func);
+    const reply = _get_func_info(req.func);
     if (req.params.read_code) {
-        const zipfile = await func_store.instance().read_code_gridfs(req.func.code_gridfs_id);
+        const system = req.system._id;
+        const name = req.params.name || _.get(req, 'params.config.name');
```
Comment (jeniawhite) [inline src/server/func_services/func_server.js:177]: This does not exist in the API
`params.config`

File: src/server/func_services/func_server.js [hunk 2]
```diff
@@ -274,16 +289,20 @@ async function invoke_func(req) {
...
+                node.name, node.pool);
+            res = await server_rpc.client.func_node.invoke_func(params, {
+                address: node.rpc_address
+            });
+        } else {
+            dbg.log0('invoking on server', func.name, req.params.event);
+            res = await server_func_node.invoke_func({ params });
+        }
+    } catch (e) {
+        dbg.error('invoke returned with error:', e);
```
Comment (jeniawhite) [inline src/server/func_services/func_server.js:305]: Add a throw here to not eat the errors.

File: src/server/func_services/func_store.js [hunk 3]
```diff
@@ -36,122 +27,77 @@ class FuncStore {
...
-            }, {
-                $set: set_updates
-            });
+    async update_func(func_id, set_updates) {
+        const res = await this._funcs.updateOne({
+            _id: func_id,
+        }, {
+            $set: set_updates
         });
+        return res;
```
Comment (jeniawhite) [inline src/server/func_services/func_store.js:56]: We shouldn't return the res of updateOne.

**Review by jeniawhite — COMMENTED** [review]
Please re-check all of the RPC APIs regarding parameters so you won't have the same issue as you've had with `params.config.`

File: src/server/func_services/func_server.js [hunk 4]
```diff
@@ -316,23 +321,19 @@ function check_event_permission(req) {
...
     if (func_code.zipfile_b64) {
         // zipfile is given as base64 string
-        return new stream.Readable({
-            read(size) {
-                this.push(Buffer.from(func_code.zipfile_b64, 'base64'));
-                this.push(null);
-            }
-        });
+        code = func_code.zipfile_b64;
+        code_size = Buffer.from(func_code.zipfile_b64).length;
```
Comment (jeniawhite) [inline src/server/func_services/func_server.js:332]: Is there a reason you did not do 'base64' or you just missed it?
Buffer.from(func_code.zipfile_b64, 'base64')

Comment (liranmauda) [reply src/server/func_services/func_server.js:332]: This flow is when the zipfile is given as base64 string

**Review by jeniawhite — APPROVED** [review]

**Pull request merged**